test/logflare/user_test.exs

```diff
@@ -0,0 +1,25 @@
+defmodule Logflare.UserTest do
```

Comment:
Tricky one...We should move `Logflare.User` to `Logflare.Users.User` which would make this move to a more appropriate module also but testing this in `Logflare.Users` is a bit on the weird side since we're testing a changeset which is not of the `Logflare.Users` concern and more from the struct itself.

Reply:
the use of `Logflare.User` instead of `Logflare.Users.User` is because this is an old codebase and didn't use much of phx codegen (or at least the standard context generation wasn't used). 
Usually changeset testing is generated to the context module's test, because outer layer modules shouldn't "reach into" the inner domain modules. The concept is from [uncle bob's clean architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html).
e.g. from the controller, we should call `Users.user_changeset` instead of `Users.User.changeset` because external users of the function shouldn't know (or even care that much) about the implementation. it makes things overly coupled with the nested context modules, making refactoring harder.
I've been keeping track of the context modules that have been refactored [here](https://github.com/Logflare/logflare/issues/1114) to match the standard generated code to make things more consistent, we'll eventually get to the User module 🏃‍♂️.

Reply:
if it's something we will tackle then I will move it 👀